sa-daily-report.rb

```diff
@@ -115,12 +117,11 @@ def scan_spam
       spam = Spam.new
       spam.filename = filename
       begin
-        Zlib::GzipReader.open(filename) { |file|
-          file.each_line { |line|
-            spam.feed(line)
+	gzip = Zlib::GzipReader.open(filename)
```

Comment:
Now that you have moved this to not use a block on `Zlib::GzipReader.open`, you are leaking a file handle as you are not closing the file handle. I would prefer this to continue using a block on `Zlib::GzipReader.open` so that the handle is handled automatically.

Reply:
I'm not really sure, if this is critical, but that change was the only solution, to get rid of this annoying warning "attempt to close unfinished zstream; reset forced." without suppressing warnings in general.

The docs show this as one of two using examples.
https://ruby-doc.org/stdlib-2.4.0/libdoc/zlib/rdoc/Zlib/GzipReader.html

Reply:
I see, GzipReader must have changed behaviour slightly from when I originally wrote this. Perhaps try calling `gzip.close` after the `.each_line` block.

Reply:
Have you tried adding this `gzip.close` call after the block yet to ensure the script is not leaking file handles?